### PR TITLE
Content attributes

### PR DESCRIPTION
**Pull request by ChaituVR**

### CODE REVIEW

**Comment by esokullu** [comment]
@ChaituVR can you fix the conflicts
then please let @ozanilbey know for him to confirm it's ok to merge,
I don't see any issue in https://github.com/phonetworks/graphjs/pull/122/files

**Comment by ChaituVR** [comment]
@ozanilbey Fixed the conflicts, :+1:

File: lib/i18n/english/tags/blog-composer.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export default {
+      "titlePlaceholder":"Enter your post title here...",
+      "saveLinkText" : "Save",
+      "savedLinkText" : "Saved",
+      "publishLinkText" : "Publish",
+      "unpublishLinkText" : "Unpublish",
+      "loginButtonText" : "Login to start a post",
+      "dummyUserText" : "Unnamed",
```
Comment (ozanilbey) [inline lib/i18n/english/tags/blog-composer.js]: "dummyUserText" : "Unnamed" is wrong. 

"Unnamed" is the value of unnamed blog post.

File: lib/i18n/english/tags/blog-composer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      "dummyContentText" : "Dummy content...",
+      "linkPromptText":"Enter the link URL",
+      "imagePromptText":"Enter the image URL",
+      "editProgessText" : "Saving...",
```
Comment (ozanilbey) [inline lib/i18n/english/tags/blog-composer.js]: "editProgessText" : "Saving..."
"editErrorText":"Couldn't be saved!"

These are wrong. This values have nothing to do with Edit option. They should be renamed as "saveProgessText", "saveErrorText"

File: lib/i18n/english/tags/blog-composer.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      "editErrorText":"Couldn't be saved!",
+      "publishProgressText" : "Publishing...",
+      "publishErrorText" : "Couldn't be published!",
+      "unpublishProgressText" : "unPublishing...",
```
Comment (ozanilbey) [inline lib/i18n/english/tags/blog-composer.js]: "unpublishProgressText" : "unPublishing..."

This is wrong. Why are you using camelCase for the text that will be read by end user. "unPublishing..." Should be "Unpublishing..."

File: lib/i18n/english/tags/blog-post.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      "replyText":"reply",
+      "repliesText":"replies",
+      "replyComposerText":"Write a Reply",
+      "replyInputPlaceholder":"Write your reply here...",
+      "replyButtonText" : "Send Reply",
+      "clearButtonText" : "Clear",
+      "loginButtonText" : "Login to write a reply",
+      "publishProgressText" : "Publishing...",
+      "publishErrorText" : "Couldn't be published!",
+      "unpublishProgressText" : "unPublishing...",
```
Comment (ozanilbey) [inline lib/i18n/english/tags/blog-post.js]: "unpublishProgressText" : "unPublishing..."

This is wrong. Why are you using camelCase for the text that will be read by end user. "unPublishing..." Should be "Unpublishing..."

File: lib/i18n/english/tags/group-header.js [hunk 5]
```diff
@@ -5,5 +5,6 @@ export default {
       "leaveGroupText" : "Leave Group",
       "groupErrorTextOne" : "Group doesn't exist.",
       "groupErrorTextTwo" : "We couldn't find any group matching this id.",
-      "refreshButtonText":"Refresh"
+      "refreshButtonText": "Refresh",
+      "saveSuccessText" : "Ok"
```
Comment (ozanilbey) [inline lib/i18n/english/tags/group-header.js]: "OK" is an abbrevation and is written in all uppercase.

File: lib/i18n/english/tags/blog-post.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      "commentTimeNowText" : "Now",
+      "commentTimeSecondsText" : "%s second(s) ago",
+      "commentTimeMinutesText" : "%s minute(s) ago",
+      "commentTimeHoursText" : "%s hour(s) ago",
+      "commentTimeDaysText" : "%s day(s) ago",
+      "commentTimeWeeksText" : "%s week(s) ago",
+      "commentTimeMonthsText" : "%s month(s) ago",
+      "commentTimeYearsText" : "%s year(s) ago",
+      "authorBeforeText" : "by",
+      "publishedTimeBeforeText" : "Published in",
```
Comment (ozanilbey) [inline lib/i18n/english/tags/blog-post.js]: Let's make this "Published on". My bad.

File: lib/styles/components/group-settings.less [hunk 7]
```diff
@@ -26,7 +26,7 @@
                 }
             }
             &::before {
-                content: "Change\00000aCover Photo";
+                content: attr(data-coverphototextlineone) " \00000a " attr(data-coverphototextlinetwo);
```
Comment (ozanilbey) [inline lib/styles/components/group-settings.less]: I am not sure that this will work. Quote characters look somehow wrong.

Comment (ozanilbey) [reply lib/styles/components/group-settings.less]: Please test this.

File: lib/styles/components/profile-settings.less [hunk 8]
```diff
@@ -26,7 +26,7 @@
                 }
             }
             &::before {
-                content: "Change\00000aAvatar";
+                content: attr(data-changeavatartextlineone) " \00000a " attr(data-changeavatartextlinetwo);
```
Comment (ozanilbey) [inline lib/styles/components/profile-settings.less]: I am not sure that this will work. Quote characters look somehow wrong.

Comment (ozanilbey) [reply lib/styles/components/profile-settings.less]: Please test this.

File: lib/tags/blog-composer.tag [hunk 9]
```diff
@@ -315,7 +320,7 @@
             if(this.validateForm) {
                 if(event) {
                     event.currentTarget.setAttribute('disabled', 'disabled');
-                    event.currentTarget.innerHTML = 'Saving...';
+                    event.currentTarget.innerHTML = i18n.editProgessText;
```
Comment (ozanilbey) [inline lib/tags/blog-composer.tag]: This should be renamed as i18n.saveProgessText

File: lib/tags/blog-composer.tag [hunk 10]
```diff
@@ -324,10 +329,10 @@
                     } else {
                         if(link) {
                             link.setAttribute('disabled', 'disabled');
-                            link.innerHTML = 'Couldn\'t be saved!';
+                            link.innerHTML = i18n.editErrorText;
```
Comment (ozanilbey) [inline lib/tags/blog-composer.tag]: This should be renamed as i18n.saveErrorText

File: lib/tags/group-creator.tag [hunk 11]
```diff
@@ -128,8 +128,10 @@
                         if(response.success) {
                             self.refs.title.classList.remove('graphjs-error');
                             self.refs.title.classList.add('graphjs-success');
+                            self.refs.title.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/group-creator.tag]: Please remove this change. self.refs.title refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-creator.tag [hunk 12]
```diff
@@ -128,8 +128,10 @@
...
                             self.refs.description.classList.remove('graphjs-error');
                             self.refs.description.classList.add('graphjs-success');
+                            self.refs.description.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/group-creator.tag]: Please remove this change. self.refs.description refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-creator.tag [hunk 13]
```diff
@@ -145,8 +147,10 @@
                             }
                         } else {
                             self.refs.title.classList.remove('graphjs-success');
+                            self.refs.title.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/group-creator.tag]: Please remove this change. self.refs.title refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-creator.tag [hunk 14]
```diff
@@ -145,8 +147,10 @@
...
                             self.refs.title.classList.add('graphjs-error');
                             self.refs.description.classList.remove('graphjs-success');
+                            self.refs.description.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/group-creator.tag]: Please remove this change. self.refs.description refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-settings.tag [hunk 15]
```diff
@@ -166,6 +166,7 @@
                                 self.group.title = title;
                                 self.refs.title.classList.remove('graphjs-error');
                                 self.refs.title.classList.add('graphjs-success');
+                                self.refs.title.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/group-settings.tag]: Please remove this change. self.refs.title refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-settings.tag [hunk 16]
```diff
@@ -175,6 +176,7 @@
                                 }
                             } else {
                                 self.refs.title.classList.remove('graphjs-success');
+                                self.refs.title.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/group-settings.tag]: Please remove this change. self.refs.title refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-settings.tag [hunk 17]
```diff
@@ -188,6 +190,7 @@
                     let successMessage = i18n.titleSuccessMessage;
                     self.refs.title.classList.remove('graphjs-error');
                     self.refs.title.classList.add('graphjs-success');
+                    self.refs.title.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/group-settings.tag]: Please remove this change. self.refs.title refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-settings.tag [hunk 18]
```diff
@@ -204,6 +207,7 @@
                                 self.group.description = description;
                                 self.refs.description.classList.remove('graphjs-error');
                                 self.refs.description.classList.add('graphjs-success');
+                                self.refs.description.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/group-settings.tag]: Please remove this change. self.refs.description refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-settings.tag [hunk 19]
```diff
@@ -213,6 +217,7 @@
                                 }
                             } else {
                                 self.refs.description.classList.remove('graphjs-success');
+                                self.refs.description.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/group-settings.tag]: Please remove this change. self.refs.description refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/group-settings.tag [hunk 20]
```diff
@@ -226,6 +231,7 @@
                     let successMessage = i18n.descriptionSuccessMessage;
                     self.refs.description.classList.remove('graphjs-error');
                     self.refs.description.classList.add('graphjs-success');
+                    self.refs.description.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/group-settings.tag]: Please remove this change. self.refs.description refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 21]
```diff
@@ -305,13 +305,15 @@
                                 self.profile.username = username;
                                 self.refs.username.classList.remove('graphjs-error');
                                 self.refs.username.classList.add('graphjs-success');
+                                self.refs.username.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.username refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 22]
```diff
@@ -305,13 +305,15 @@
...
                                 self.failMessages.includes(failMessage) && self.failMessages.splice(self.failMessages.indexOf(failMessage), 1);
                                 self.successMessages.includes(successMessage) || self.successMessages.push(successMessage);
                                 self.refs.submitProfile.classList.remove('graphjs-loading');
                                 self.update();
                                 self.parent.tags.hasOwnProperty('graphjs-profile-header') && self.parent.tags['graphjs-profile-header'].updateInformation();
                             } else {
                                 self.refs.username.classList.remove('graphjs-success');
+                                self.refs.username.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.username refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 23]
```diff
@@ -325,6 +327,7 @@
                     let successMessage = i18n.usernameSuccessMessage;
                     self.refs.username.classList.remove('graphjs-error');
                     self.refs.username.classList.add('graphjs-success');
+                    self.refs.username.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.username refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 24]
```diff
@@ -340,13 +343,15 @@
                                 self.profile.email = email;
                                 self.refs.email.classList.remove('graphjs-error');
                                 self.refs.email.classList.add('graphjs-success');
+                                self.refs.email.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.email refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 25]
```diff
@@ -340,13 +343,15 @@
...
                                 self.failMessages.includes(failMessage) && self.failMessages.splice(self.failMessages.indexOf(failMessage), 1);
                                 self.successMessages.includes(successMessage) || self.successMessages.push(successMessage);
                                 self.refs.submitProfile.classList.remove('graphjs-loading');
                                 self.update();
                                 self.parent.tags.hasOwnProperty('graphjs-profile-header') && self.parent.tags['graphjs-profile-header'].updateInformation();
                             } else {
                                 self.refs.email.classList.remove('graphjs-success');
+                                self.refs.email.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.email refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 26]
```diff
@@ -360,6 +365,7 @@
                     let successMessage = i18n.emailSuccessMessage;
                     self.refs.email.classList.remove('graphjs-error');
                     self.refs.email.classList.add('graphjs-success');
+                    self.refs.email.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.email refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 27]
```diff
@@ -374,13 +380,15 @@
                             self.profile.about = bio;
                             self.refs.bio.classList.remove('graphjs-error');
                             self.refs.bio.classList.add('graphjs-success');
+                            self.refs.bio.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.bio refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 28]
```diff
@@ -374,13 +380,15 @@
...
                             self.failMessages.includes(failMessage) && self.failMessages.splice(self.failMessages.indexOf(failMessage), 1);
                             self.successMessages.includes(successMessage) || self.successMessages.push(successMessage);
                             self.refs.submitProfile.classList.remove('graphjs-loading');
                             self.update();
                             self.parent.tags.hasOwnProperty('graphjs-profile-header') && self.parent.tags['graphjs-profile-header'].updateInformation();
                         } else {
                             self.refs.bio.classList.remove('graphjs-success');
+                            self.refs.bio.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.bio refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 29]
```diff
@@ -398,13 +406,15 @@
                             self.profile.birthday = birthday;
                             self.refs.birthday.classList.remove('graphjs-error');
                             self.refs.birthday.classList.add('graphjs-success');
+                            self.refs.birthday.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.birthday refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 30]
```diff
@@ -398,13 +406,15 @@
...
                             self.failMessages.includes(failMessage) && self.failMessages.splice(self.failMessages.indexOf(failMessage), 1);
                             self.successMessages.includes(successMessage) || self.successMessages.push(successMessage);
                             self.refs.submitProfile.classList.remove('graphjs-loading');
                             self.update();
                             self.parent.tags.hasOwnProperty('graphjs-profile-header') && self.parent.tags['graphjs-profile-header'].updateInformation();
                         } else {
                             self.refs.birthday.classList.remove('graphjs-success');
+                            self.refs.birthday.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.birthday refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 31]
```diff
@@ -435,14 +445,18 @@
                             self.refs.password.classList.remove('graphjs-error');
                             self.refs.confirmation.classList.remove('graphjs-error');
                             self.refs.password.classList.add('graphjs-success');
+                            self.refs.password.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.password refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 32]
```diff
@@ -435,14 +445,18 @@
...
                             self.refs.confirmation.classList.add('graphjs-success');
+                            self.refs.confirmation.setAttribute('data-savesuccesstext', i18n.saveSuccessText);
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.confirmation refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 33]
```diff
@@ -435,14 +445,18 @@
...
                             self.failMessages.includes(failMessage) && self.failMessages.splice(self.failMessages.indexOf(failMessage), 1);
                             self.successMessages.includes(successMessage) || self.successMessages.push(successMessage);
                             self.refs.submitPassword.classList.remove('graphjs-loading');
                             self.update();
                         } else {
                             self.refs.password.classList.remove('graphjs-success');
                             self.refs.confirmation.classList.remove('graphjs-success');
+                            self.refs.password.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.password refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/tags/profile-settings.tag [hunk 34]
```diff
@@ -435,14 +445,18 @@
...
+                            self.refs.confirmation.removeAttribute('data-savesuccesstext');
```
Comment (ozanilbey) [inline lib/tags/profile-settings.tag]: Please remove this change. self.refs.confirmation refers to an input element and input elements do not accept ::before or ::after pseudo-elements.

File: lib/styles/components/group-settings.less (hunk 7 above)
Comment (ChaituVR) [reply lib/styles/components/group-settings.less]: Tested Working fine 
image

File: lib/styles/components/profile-settings.less (hunk 8 above)
Comment (ChaituVR) [reply lib/styles/components/profile-settings.less]: Tested Working Fine

File: lib/tags/group-creator.tag (hunk 11 above)
Comment (ChaituVR) [reply lib/tags/group-creator.tag]: i've added data-attribute where ever there is `content` in css, we can remove that line but
as pseudo-elements are not supported for input we may have to remove all css in all .less files right?? which is not related to this PR.. should just remove this line for now?

File: lib/tags/group-creator.tag (hunk 12 above)
Comment (ChaituVR) [reply lib/tags/group-creator.tag]: Same as above

File: lib/tags/group-creator.tag (hunk 13 above)
Comment (ChaituVR) [reply lib/tags/group-creator.tag]: Same as above

File: lib/tags/group-creator.tag (hunk 14 above)
Comment (ChaituVR) [reply lib/tags/group-creator.tag]: Same as above

File: lib/tags/group-settings.tag (hunk 15 above)
Comment (ChaituVR) [reply lib/tags/group-settings.tag]: Same as above

File: lib/tags/group-settings.tag (hunk 16 above)
Comment (ChaituVR) [reply lib/tags/group-settings.tag]: Same as above

File: lib/tags/group-settings.tag (hunk 17 above)
Comment (ChaituVR) [reply lib/tags/group-settings.tag]: Same as above

File: lib/tags/group-settings.tag (hunk 18 above)
Comment (ChaituVR) [reply lib/tags/group-settings.tag]: Same as above

File: lib/tags/group-settings.tag (hunk 19 above)
Comment (ChaituVR) [reply lib/tags/group-settings.tag]: Same as above

File: lib/tags/group-settings.tag (hunk 20 above)
Comment (ChaituVR) [reply lib/tags/group-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 21 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 22 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 23 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 24 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 25 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 26 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 27 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 28 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 29 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 30 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 32 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 33 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/tags/profile-settings.tag (hunk 34 above)
Comment (ChaituVR) [reply lib/tags/profile-settings.tag]: Same as above

File: lib/i18n/english/tags/blog-post.js [hunk 35]
```diff
@@ -31,6 +31,6 @@ export default {
       "commentTimeMonthsText" : "%s month(s) ago",
       "commentTimeYearsText" : "%s year(s) ago",
       "authorBeforeText" : "by",
-      "publishedTimeBeforeText" : "Published in",
+      "publishedTimeBeforeText" : "Published On",
```
Comment (ozanilbey) [inline lib/i18n/english/tags/blog-post.js]: This should be "Published on", not "Published On"

**Pull request merged**